Tests/Domain/RoleSecurityIdentityTest.php

```diff
@@ -47,7 +51,6 @@ public function getCompareData()
     {
         return [
             [new RoleSecurityIdentity('ROLE_FOO'), new RoleSecurityIdentity('ROLE_FOO'), true],
-            [new RoleSecurityIdentity('ROLE_FOO'), new RoleSecurityIdentity(new Role('ROLE_FOO')), true],
```

Comment:
this case should still be re-added as a separate test that can be skipped, or added dynamically in the data provider

Reply:
I've added a separate test that is marked as `legacy` and that is skipped if the role class is not found